### DISCUSSION TITLE
Customizing Highlighted Colors

### DISCUSSION THREAD

**Discussion by wgmayer0** [qutebrowser/qutebrowser#5978] [Q&A]
Hello,

First of all, The-Compiler did an incredible job on Qutebrowser. It is one of the best browsers ever written and works very well.

Something I have been wanting to do is change the color of search results as well as the highlighted colors.

By default, Qutebrowser uses these settings:

The active search result: orange
The other non active search results: yellow
The cursor highlighted color: blue

I would think this has something to do with Qt but how do I customize these colors?

Thanks,

**Comment by The-Compiler** [comment]
I've deleted #5975 and #5976 since you've opened the same discussion three times.

As far as I know, those colors are hardcoded in the underlying Chromium and can't be changed - except for the selected text color maybe, that might be possible to do via CSS and `content.user_stylesheets` somehow.

**Reply by wgmayer0** [reply]
Does this help at all? https://github.com/Andreto/Custom-Highlight/issues/8#issuecomment-749973358

**Reply by rien333** [reply]
> As far as I know, those colors are hardcoded in the underlying Chromium and can't be changed

FWIW, this is true. I build qt5-webengine from source, and because I don't like how the default highlight colors pair with darkmode, I thought I might as well change them. 

Applying this patch to the code allows one to change the highlight colors (though it should be noted that building qt5-webengine from source is quite a lengthy process):

```diff
diff --git a/src/3rdparty/chromium/third_party/blink/renderer/core/layout/layout_theme.cc b/src/3rdparty/chromium/third_party/blink/renderer/core/layout/layout_theme.cc
index 488f6f7..9567662 100644
--- a/src/3rdparty/chromium/third_party/blink/renderer/core/layout/layout_theme.cc
+++ b/src/3rdparty/chromium/third_party/blink/renderer/core/layout/layout_theme.cc
@@ -824,9 +824,9 @@ Color LayoutTheme::PlatformTextSearchHighlightColor(
   if (active_match) {
     if (in_forced_colors_mode)
       return GetTheme().SystemColor(CSSValueID::kHighlight, color_scheme);
-    return Color(255, 150, 50);  // Orange.
+    return Color(67,147,206);  // Orange.
   }
-  return Color(255, 255, 0);     // Yellow.
+  return Color(168,190,212);     // Yellow.
 }
 
 Color LayoutTheme::PlatformTextSearchColor(bool active_match,
 ```

**Reply by wgmayer0** [reply]
Thanks this looks like it would do it. Do you have a link to a guide or some reading I can follow to learn how to build from source? Also, this is a more general question, but if I build from source does that mean I won't be able to update Qutebrowser?

**Reply by rien333** [reply]
> Do you have a link to a guide or some reading I can follow to learn how to build from source?

This will depend on your distro a little. On arch linux, I just modify the PKGBUILD (a recipe or list of commands that build a package)  so that it includes this patch. On debian-based distros, you could probably also take a look at how maintainers build packages, and modify that accordingly (though when I look at the websites of ubuntu/debian, it's not immediately clear to me how they build qt5-webengine). Still, if this seems like a little too much to you, this is probably not worth it at this point for this feature alone. Another sort of reference would be: https://wiki.qt.io/QtWebEngine/How_to_Try

> does that mean I won't be able to update Qutebrowser?

No, qt5-webengine and qutebrowser are somewhat independent of one another. You can still receive qutebrowser updates from your package manager if  you have webengine build from source.  Updating qt5-webengine, on the other hand, will be a little more  hassle than simply running an "upgrade" using your package manager.

**Reply by wgmayer0** [reply]
Ok yeah that is what I meant, update qt5-webengine, thanks. I will look into custom modifications of source code on Arch packages. Thanks again for the thorough answer!

**Comment by funcrab** [comment]
@wgmayer0 
> Something I have been wanting to do is change the color of search results as well as the highlighted colors.

I changed the theme settings of the search engine to make it easier to see the color of the links in the search engine, which is a quick fix and limited to google, ddg, etc.
Unfortunately, other methods are still too difficult for me, so I gave up.

**Reply by The-Compiler** [reply]
@funcrab note that @wgmayer0 was talking about the "find on page" colors you get with e.g. `/test`, not a search engine theme.

**Reply by funcrab** [reply]
@The-Compiler Thank you. I was mistaken.
